code/__HELPERS/_logging.dm

```diff
@@ -149,8 +149,19 @@ GLOBAL_PROTECT(log_end)
 /proc/log_runtime_summary(text)
 	rustg_log_write(GLOB.runtime_summary_log, "[text][GLOB.log_end]")
 
-/proc/log_tgui(text)
-	rustg_log_write(GLOB.tgui_log, "[text][GLOB.log_end]")
+/proc/log_tgui(user_or_client, text)
+	var/entry = ""
+	if(!user_or_client)
+		entry += "no user"
+	else if(istype(user_or_client, /mob))
+		var/mob/user = user_or_client
+		entry += "[user.ckey] (as [user])"
+	else if(istype(user_or_client, /client))
+		var/client/client = user_or_client
+		entry += "[client.ckey]"
+	entry += ":\n[text]"
+	entry += "[GLOB.log_end]"
+	rustg_log_write(GLOB.tgui_log, entry)
```

Comment:
It might actually be better to convert this to a list operation as opposed to a string operation. An example of how to do this can be found in `/mob/living/simple_animal/revenant/proc/giveObjectivesandGoals()`
https://github.com/ParadiseSS13/Paradise/blob/ff53bdd87ff6500f06e0997ba3f4a16151e08778/code/game/gamemodes/miniantags/revenant/revenant.dm#L176-L192


</p>
</details>

Reply:
Updated with suggestion below and some additional changes. Tested for proper behavior.